packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -253,41 +253,47 @@ export class ListLayout<T, O extends ListLayoutOptions = ListLayoutOptions> exte
 
   protected buildCollection(y = this.padding): LayoutNode[] {
     let collection = this.virtualizer!.collection;
-    let skipped = 0;
+    let collectionNodes = [...collection];
+    let loaderNodes = collectionNodes.filter(node => node.type === 'loader');
```

Comment:
darn, need to wait a year until we can use https://caniuse.com/?search=Iterator.prototype.filter to avoid an extra traversal